tests/test_pf7.py

```diff
@@ -0,0 +1,433 @@
+            "FILTER_VQSRTrancheSNP99.50to99.60": [DIM_VARIANT],
+            "FILTER_VQSRTrancheSNP99.60to99.80": [DIM_VARIANT],
+            "FILTER_VQSRTrancheSNP99.80to99.90": [DIM_VARIANT],
+            "FILTER_VQSRTrancheSNP99.90to99.95": [DIM_VARIANT],
+            "FILTER_VQSRTrancheSNP99.95to100.00+": [DIM_VARIANT],
+            "FILTER_VQSRTrancheSNP99.95to100.00": [DIM_VARIANT],
+            "FS": [DIM_VARIANT],
+            "ID": [DIM_VARIANT],
+            "InbreedingCoeff": [DIM_VARIANT],
+            "LOF": [DIM_VARIANT],
+            "MLEAC": [DIM_VARIANT, DIM_ALT_ALLELE],
+            "MLEAF": [DIM_VARIANT, DIM_ALT_ALLELE],
+            "MQ": [DIM_VARIANT],
+            "MQRankSum": [DIM_VARIANT],
+            "NEGATIVE_TRAIN_SITE": [DIM_VARIANT],
+            "NMD": [DIM_VARIANT],
+            "POSITIVE_TRAIN_SITE": [DIM_VARIANT],
+            "QD": [DIM_VARIANT],
+            "QUAL": [DIM_VARIANT],
+            "RAW_MQandDP": [DIM_VARIANT, DIM_PLOIDY],
+            "ReadPosRankSum": [DIM_VARIANT],
+            "RegionType": [DIM_VARIANT],
+            "SOR": [DIM_VARIANT],
+            "VQSLOD": [DIM_VARIANT],
+            "altlen": [DIM_VARIANT, DIM_ALT_ALLELE],
+            "culprit": [DIM_VARIANT],
+            "set": [DIM_VARIANT],
+        }
+
+    @patch("malariagen_data.pf7.init_filesystem", return_value=["fs", "path"])
+    def test_setup_with_config_none(self, mock_init_filesystem):
+        with patch(
+            "malariagen_data.pf7.Pf7._load_config",
+            return_value=self.config_content,
+        ) as mock_load_config:
+            pf7 = Pf7(self.url_starts_with_gs)
+        self.assertEqual(pf7._cache_sample_metadata, None)
+        self.assertEqual(pf7._cache_zarr, None)
+        self.assertEqual(
+            pf7.extended_calldata_variables, self.test_extended_calldata_variables
+        )
+        self.assertEqual(pf7.extended_variant_fields, self.test_extended_variant_fields)
+        mock_load_config.assert_called_once_with(None)
+        mock_init_filesystem.assert_called_once_with(self.url_starts_with_gs)
+
+    @patch("malariagen_data.pf7.init_filesystem", return_value=["fs", "path"])
+    def test_setup_with_config_set(self, mock_init_filesystem):
+        with patch(
+            "malariagen_data.pf7.Pf7._load_config",
+            return_value=self.config_content,
+        ) as mock_load_config:
+            pf7 = Pf7(self.url_starts_with_gs, data_config=self.test_config_path)
+        self.assertEqual(pf7._cache_sample_metadata, None)
+        self.assertEqual(pf7._cache_zarr, None)
+        self.assertEqual(
+            pf7.extended_calldata_variables, self.test_extended_calldata_variables
+        )
+        self.assertEqual(pf7.extended_variant_fields, self.test_extended_variant_fields)
+        mock_load_config.assert_called_once_with(self.test_config_path)
+        mock_init_filesystem.assert_called_once_with(self.url_starts_with_gs)
```

Comment:
Excess baggage?

Reply:
I have kept this test but removed a lot of the unnecessary asserts and mocking. I am just using this to check the config is loaded correctly now